x-pack/plugins/security_solution/public/common/hooks/eql/helpers.ts

```diff
@@ -0,0 +1,190 @@
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import moment from 'moment';
+import { Unit } from '@elastic/datemath';
+
+import * as i18n from '../../../detections/components/rules/query_preview/translations';
+import { EqlSearchStrategyResponse } from '../../../../../data_enhanced/common';
+import { InspectResponse } from '../../../types';
+import { EqlPreviewResponse, Source } from './types';
+import { EqlSearchResponse } from '../../../../common/detection_engine/types';
+import { HITS_THRESHOLD } from '../../../detections/components/rules/query_preview/helpers';
+
+type EqlAggBuckets = Record<string, { timestamp: string; total: number }>;
+
+export const EQL_QUERY_EVENT_SIZE = 100;
+
+// Calculates which 2 min bucket segment, event should be
+// sorted into
+export const calculateBucketForHour = (eventTimestamp: number, relativeNow: number): number => {
+  const diff: number = relativeNow - eventTimestamp;
+  const minutes: number = Math.floor(diff / 60000);
+  return Math.ceil(minutes / 2) * 2;
+};
+
+// Calculates which 1 hour bucket segment, event should be
+// sorted into
+export const calculateBucketForDay = (eventTimestamp: number, relativeNow: number): number => {
+  const diff: number = relativeNow - eventTimestamp;
+  const minutes: number = Math.floor(diff / 60000);
+  return Math.ceil(minutes / 60);
+};
+
+export const constructWarnings = (timestampIssue: boolean, hits: number, range: Unit): string[] => {
+  let warnings: string[] = [];
+
+  if (timestampIssue) {
+    warnings = [i18n.PREVIEW_WARNING_TIMESTAMP];
+  }
+
+  if (hits === EQL_QUERY_EVENT_SIZE) {
+    warnings = [...warnings, i18n.PREVIEW_WARNING_CAP_HIT(EQL_QUERY_EVENT_SIZE)];
+  }
+
+  if (hits > HITS_THRESHOLD[range]) {
+    warnings = [...warnings, i18n.QUERY_PREVIEW_NOISE_WARNING];
+  }
+
+  return warnings;
+};
+
+export const formatInspect = (
+  response: EqlSearchStrategyResponse<EqlSearchResponse<Source>>
+): InspectResponse => {
+  if (response != null) {
+    return {
+      dsl: [JSON.stringify(response.rawResponse.meta.request.params, null, 2)] ?? [],
+      response: [JSON.stringify(response.rawResponse.body, null, 2)] ?? [],
+    };
+  }
+
+  return {
+    dsl: [],
+    response: [],
+  };
+};
+
+// NOTE: Eql does not support aggregations, this is an in-memory
+// hand-spun aggregation for the events to give the user a visual
+// representation of their query results
+export const getEqlAggsData = (
+  response: EqlSearchStrategyResponse<EqlSearchResponse<Source>>,
+  range: Unit,
+  to: string,
+  from: string
+): EqlPreviewResponse => {
+  const { dsl, response: inspectResponse } = formatInspect(response);
+  // The upper bound of the timestamps
+  const relativeNow: number = Date.parse(from);
+  const accumulator: EqlAggBuckets = getInterval(range, relativeNow);
```

Comment:
nit: You don't need to add types here as both these functions return types correctly.

Reply:
Removed here: https://github.com/elastic/kibana/pull/79947